### PR TITLE
Codecoverage(libc/stdio): Add TC's for few API's in stdio

### PR DESCRIPTION
**Pull request by thapav**
This patch adds test cases for stream initialization API's namely:
memoutstream, meminstream, memsistream, memsostream,
rawinstream, rawoutstream, rawsistream, rawsostream,
stdinstream, stdoutstream, stdsistream, stdsostream
nullinstream, nulloutstream, zeroinstream

Signed-off-by: Vidisha Thapa <thapa.v@samsung.com>

### CODE REVIEW

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 1]
```diff
@@ -1961,6 +1965,400 @@ static void libc_stdio_setvbuf_tc(void)
...
+*/
+static void libc_stdio_memoutstream_tc(void)
+{
+	FAR char buf[BUFLEN];
+
+	struct lib_memoutstream_s memoutstream;
+
+	lib_memoutstream((FAR struct lib_memoutstream_s *)&memoutstream, buf, BUFLEN);
+	TC_ASSERT_EQ("lib_memoutstream", memoutstream.buffer, (FAR char *)(buf));
+	TC_ASSERT_EQ("lib_memoutstream", memoutstream.buflen, (BUFLEN-1));	/* Save space for null terminator, hence checing with (BUFLEN-1)*/
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: add space between operand and operator

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 2]
```diff
@@ -1961,6 +1965,400 @@ static void libc_stdio_setvbuf_tc(void)
...
+{
+	FAR char buf[BUFLEN];
+
+	struct lib_meminstream_s meminstream;
+	struct lib_memsistream_s memsistream;
+
+	lib_meminstream((FAR struct lib_meminstream_s *)&meminstream, buf, BUFLEN);
+	TC_ASSERT_EQ("lib_meminstream", meminstream.buffer, (FAR char *)(buf));
+
+	lib_memsistream((FAR struct lib_memsistream_s *)&memsistream, buf+2, BUFLEN);
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: same as above

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 3]
```diff
@@ -1961,6 +1965,400 @@ static void libc_stdio_setvbuf_tc(void)
...
+{
+	FAR char buf[BUFLEN];
+
+	struct lib_memoutstream_s memoutstream;
+	struct lib_memsostream_s memsostream;
+
+	lib_memoutstream((FAR struct lib_memoutstream_s *)&memoutstream, buf, BUFLEN);
+	TC_ASSERT_EQ("lib_memoutstream", memoutstream.buffer, (FAR char *)(buf));
+
+	lib_memsostream((FAR struct lib_memsostream_s *)&memsostream, buf+4, BUFLEN);
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: same as above

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 4]
```diff
@@ -1961,6 +1965,400 @@ static void libc_stdio_setvbuf_tc(void)
...
+* @apicovered       lib_nullinstream
+* @precondition     NA
+* @postcondition    NA
+*/
+static void libc_stdio_nullinstream_tc(void)
+{
+        union {
+                struct lib_instream_s nullinstream;
+                struct lib_meminstream_s meminstream;
+        } u;
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: please declare the variable with clear name

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 5]
```diff
@@ -1961,6 +1965,400 @@ static void libc_stdio_setvbuf_tc(void)
...
+	FILE *fd;
+	char *filename = VFS_FILE_PATH;
+
+	struct lib_rawoutstream_s rawoutstream;
+
+	fd = fopen(filename, "w");
+	TC_ASSERT("fopen", fd);
+	fclose(fd);
+
+	lib_rawoutstream((FAR struct lib_rawoutstream_s *)&rawoutstream, (int)fd);
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: in line 2158, you closed the fd, but you used that fd in this line.
is it okay?

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 6]
```diff
@@ -1961,6 +1965,400 @@ static void libc_stdio_setvbuf_tc(void)
...
+	struct lib_rawinstream_s rawinstream;
+	struct lib_rawsistream_s rawsistream;
+
+	fd = fopen(filename, "w");
+	TC_ASSERT("fopen", fd);
+
+	lib_rawinstream((FAR struct lib_rawinstream_s *)&rawinstream, (int)fd);
+	TC_ASSERT_EQ_CLEANUP("lib_rawinstream", rawinstream.public.nget, 0, fclose(fd));
+
+	lib_rawsistream((FAR struct lib_rawsistream_s *)&rawsistream, (int)(fd+2));
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: add space between operand and operator

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 7]
```diff
@@ -1961,6 +1965,400 @@ static void libc_stdio_setvbuf_tc(void)
...
+	struct lib_rawsostream_s rawsostream;
+
+	fd = fopen(filename, "w");
+	TC_ASSERT("fopen", fd);
+	fclose(fd);
+
+	lib_rawoutstream((FAR struct lib_rawoutstream_s *)&rawoutstream, (int)fd);
+	TC_ASSERT_EQ_CLEANUP("lib_rawoutstream", rawoutstream.public.nput, 0, fclose(fd));
+
+	lib_rawsostream((FAR struct lib_rawsostream_s *)&rawsostream, (int)(fd+4));
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: add space

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 8]
```diff
@@ -1961,6 +1965,400 @@ static void libc_stdio_setvbuf_tc(void)
...
+	struct lib_stdoutstream_s stdoutstream;
+	struct lib_stdsostream_s stdsostream;
+
+	fd = fopen(filename, "w");
+	TC_ASSERT("fopen", fd);
+
+	lib_stdoutstream((FAR struct lib_stdoutstream_s *)&stdoutstream, fd);
+	TC_ASSERT_EQ_CLEANUP("lib_stdoutstream", stdoutstream.public.nput, 0, fclose(fd));
+
+	lib_stdsostream((FAR struct lib_stdsostream_s *)&stdsostream, (fd+2));
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: add space

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 9]
```diff
@@ -39,7 +39,10 @@
 #include <sys/sendfile.h>
 #include <sys/statfs.h>
 #include <sys/select.h>
+#include <sys/types.h>
+#include "../../../../../lib/libc/lib_internal.h"
 
+#include <tinyara/config.h>
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: please move this include to the top

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 10]
```diff
@@ -1961,6 +1965,400 @@ static void libc_stdio_setvbuf_tc(void)
...
+	struct lib_stdinstream_s stdinstream;
+	struct lib_stdsistream_s stdsistream;
+
+	fd = fopen(filename, "w");
+	TC_ASSERT("fopen", fd);
+
+	lib_stdinstream((FAR struct lib_stdinstream_s *)&stdinstream, fd);
+	TC_ASSERT_EQ_CLEANUP("lib_stdinstream", stdinstream.public.nget, 0, fclose(fd));
+
+	lib_stdsistream((FAR struct lib_stdsistream_s *)&stdsistream, (fd+2));
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: add space

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 5 above)
Comment (thapav) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: no, will take care of that, left out mistakenly

**Comment by thapav** [comment]
modified as per review comments

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 11]
```diff
@@ -1961,6 +1965,398 @@ static void libc_stdio_setvbuf_tc(void)
...
+
+	struct lib_memoutstream_s memoutstream;
+	struct lib_memsostream_s memsostream;
+
+	lib_memoutstream((FAR struct lib_memoutstream_s *)&memoutstream, buf, BUFLEN);
+	TC_ASSERT_EQ("lib_memoutstream", memoutstream.buffer, (FAR char *)(buf));
+
+	lib_memsostream((FAR struct lib_memsostream_s *)&memsostream, buf + 4, BUFLEN);
+	TC_ASSERT_EQ("lib_memsostream", (memsostream.buffer - (FAR char *)(buf)), 4);
+	TC_ASSERT_EQ("lib_memsostream", memsostream.buflen, (BUFLEN-1));	/* Save space for null terminator, hence checing with (BUFLEN-1)*/
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: space

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 12]
```diff
@@ -1961,6 +1965,398 @@ static void libc_stdio_setvbuf_tc(void)
...
+	FILE *fd;
+	char *filename = VFS_FILE_PATH;
+
+	struct lib_rawoutstream_s rawoutstream;
+
+	fd = fopen(filename, "w");
+	TC_ASSERT("fopen", fd);
+
+	lib_rawoutstream((FAR struct lib_rawoutstream_s *)&rawoutstream, (int)fd);
+	TC_ASSERT_EQ_CLEANUP("lib_rawoutstream", rawoutstream.public.nput, 0, fclose(fd));
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: Your codes
```
if (xxx != 0) {
  .....
  fclose(fd);
  return;
}

fclose(fd);
```

fclose(fd) should be called always.
Let's change like below to reduce code size.
```
fclose(fd);
TC_ASSERT_EQ(xxxx, yyyy, 0);
```

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 13]
```diff
@@ -1961,6 +1965,398 @@ static void libc_stdio_setvbuf_tc(void)
...
+
+	fd = fopen(filename, "w");
+	TC_ASSERT("fopen", fd);
+
+	lib_rawinstream((FAR struct lib_rawinstream_s *)&rawinstream, (int)fd);
+	TC_ASSERT_EQ_CLEANUP("lib_rawinstream", rawinstream.public.nget, 0, fclose(fd));
+
+	lib_rawsistream((FAR struct lib_rawsistream_s *)&rawsistream, (int)(fd + 2));
+	TC_ASSERT_EQ_CLEANUP("lib_rawsistream", ((FAR FILE *)rawsistream.fd - fd), 2, fclose(fd));
+	TC_ASSERT_EQ_CLEANUP("lib_rawsistream", rawsistream.public.nget, 0, fclose(fd));
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: same as previous

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 14]
```diff
@@ -1961,6 +1965,398 @@ static void libc_stdio_setvbuf_tc(void)
...
+
+	fd = fopen(filename, "w");
+	TC_ASSERT("fopen", fd);
+
+	lib_rawoutstream((FAR struct lib_rawoutstream_s *)&rawoutstream, (int)fd);
+	TC_ASSERT_EQ_CLEANUP("lib_rawoutstream", rawoutstream.public.nput, 0, fclose(fd));
+
+	lib_rawsostream((FAR struct lib_rawsostream_s *)&rawsostream, (int)(fd + 4));
+	TC_ASSERT_EQ_CLEANUP("lib_rawsostream", ((FAR FILE *)rawsostream.fd - fd), 4, fclose(fd));
+	TC_ASSERT_EQ_CLEANUP("lib_rawsostream", rawsostream.public.nput, 0, fclose(fd));
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: same, there are many codes like this. Please check all.

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 12 above)
Comment (thapav) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: done

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 15]
```diff
@@ -39,7 +39,10 @@
 #include <sys/sendfile.h>
 #include <sys/statfs.h>
 #include <sys/select.h>
+#include <sys/types.h>
+#include "../../../../../lib/libc/lib_internal.h"
```
[the remark this answers is not in the document]
Comment (thapav) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: added <streams.h> instead

**Comment by sunghan-chang** [comment]
You missed many places for my previous comment.
```
fclose(fd);
TC_ASSERT_EQ(xxxx, yyyy, 0);
```

**Comment by thapav** [comment]
@sunghan-chang ,,,just added them, please check now

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 16]
```diff
@@ -2068,6 +2456,21 @@ static int fs_sample_launcher(int argc, char **args)
...
+	libc_stdio_nulloutstream_tc();
+	libc_stdio_rawinstream_tc();
+	libc_stdio_rawoutstream_tc();
+	libc_stdio_rawsistream_tc();
+	libc_stdio_rawsostream_tc();
+	libc_stdio_stdinstream_tc();
+	libc_stdio_stdoutstream_tc();
+	libc_stdio_stdsistream_tc();
+	libc_stdio_stdsostream_tc();
+	libc_stdio_zeroinstream_tc();
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: please check the alphabet sequence

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 17]
```diff
@@ -1961,6 +1965,390 @@ static void libc_stdio_setvbuf_tc(void)
...
+*/
+static void libc_stdio_rawinstream_tc(void)
+{
+	FILE *fd;
+	char *filename = VFS_FILE_PATH;
+
+	struct lib_rawinstream_s rawinstream;
+
+	fd = fopen(filename, "w");
+	TC_ASSERT("fopen", fd);
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: I gave my opinion for this to @jeongarmy and @jeongchanKim .
TC_ASSERT(xxx, yy) and TC_ASSERT_NEQ(xxx, yy, NULL or false) are same.
Currently, some TC uses TC_ASSERT, some uses TC_ASSERT_NEQ.
I want to keep same policy for readability.

Comment (jeongchanKim) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: We will not use TC_ASSERT() anymore.
so please replace TC_ASSERT() to proper macro

Comment (thapav) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: done

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 16 above)
Comment (thapav) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: it seems correct

Comment (jeongchanKim) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: libc_stdio_zeroinstream_tc should be placed after libc_stdio_ungetc_tc

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 18]
```diff
@@ -1961,6 +1965,390 @@ static void libc_stdio_setvbuf_tc(void)
...
+* @postcondition    NA
+*/
+static void libc_stdio_memsistream_tc(void)
+{
+	FAR char buf[BUFLEN];
+
+	struct lib_meminstream_s meminstream;
+	struct lib_memsistream_s memsistream;
+
+	lib_meminstream((FAR struct lib_meminstream_s *)&meminstream, buf, BUFLEN);
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: Why did you add these codes in libc_stdio_meminstream_tc here again?

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 19]
```diff
@@ -1961,6 +1965,390 @@ static void libc_stdio_setvbuf_tc(void)
...
+	TC_ASSERT_EQ("lib_meminstream", meminstream.buffer, (FAR char *)(buf));
+
+	lib_memsistream((FAR struct lib_memsistream_s *)&memsistream, buf + 2, BUFLEN);
+	TC_ASSERT_EQ("lib_memsistream", (memsistream.buffer - (FAR char *)(buf)), 2);
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: what means 2? just any number?

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 20]
```diff
@@ -1961,6 +1965,390 @@ static void libc_stdio_setvbuf_tc(void)
...
+*/
+static void libc_stdio_memsostream_tc(void)
+{
+	FAR char buf[BUFLEN];
+
+	struct lib_memoutstream_s memoutstream;
+	struct lib_memsostream_s memsostream;
+
+	lib_memoutstream((FAR struct lib_memoutstream_s *)&memoutstream, buf, BUFLEN);
+	TC_ASSERT_EQ("lib_memoutstream", memoutstream.buffer, (FAR char *)(buf));
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: same above. Why did you add these codes for memoutstream here again?

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 21]
```diff
@@ -1961,6 +1965,390 @@ static void libc_stdio_setvbuf_tc(void)
...
+{
+        union {
+                struct lib_outstream_s nulloutstream;
+                struct lib_memoutstream_s memoutstream;
+        } outstream_object;
+
+	FAR char buf[BUFLEN];
+	FAR struct lib_outstream_s *stream;
+
+	lib_memoutstream((FAR struct lib_memoutstream_s *)&outstream_object.memoutstream, buf, BUFLEN);
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: same above. I don't understand these codes are needed for this tc.

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 22]
```diff
@@ -1961,6 +1965,390 @@ static void libc_stdio_setvbuf_tc(void)
...
+*/
+static void libc_stdio_rawinstream_tc(void)
+{
+	FILE *fd;
+	char *filename = VFS_FILE_PATH;
+
+	struct lib_rawinstream_s rawinstream;
+
+	fd = fopen(filename, "w");
+	TC_ASSERT_NEQ("fopen", fd, OK);
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: I think this should be TC_ASSERT_NEQ(fd, NULL);, not OK.
fopen returns pointer or NULL.

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 23]
```diff
@@ -1961,6 +1965,390 @@ static void libc_stdio_setvbuf_tc(void)
...
+
+	lib_rawinstream((FAR struct lib_rawinstream_s *)&rawinstream, (int)fd);
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: You got the file pointer and changed type of it to get fd value.
Is it correct? It looks wrong.
We don't need to get fd using open?

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 24]
```diff
@@ -1961,6 +1965,390 @@ static void libc_stdio_setvbuf_tc(void)
...
+	fclose(fd);
+	TC_ASSERT_EQ("lib_rawinstream", rawinstream.public.nget, 0);
+
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: I think that it is needed to check whether rawinstream.fd is fd or not.

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 25]
```diff
@@ -1961,6 +1965,390 @@ static void libc_stdio_setvbuf_tc(void)
...
+/**
+* @testcase         libc_stdio_rawoutstream_tc
+* @brief            Initializes a stream for use with a file descriptor
+* @scenario         Initializes a stream for use with a file descriptor
+* @apicovered       lib_rawoutstream
+* @precondition     NA
+* @postcondition    NA
+*/
+static void libc_stdio_rawoutstream_tc(void)
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: Same two comments in lib_stdio_rawinstream_tc

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 16 above)
Comment (thapav) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: ok will do that, sorry did not spot that

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 19 above)
Comment (thapav) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: Yes , just to seek through that stream and check if its seeking correctly

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 21 above)
Comment (thapav) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: Yes i crosschecked, i think we dont need it for nullstreams

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 20 above)
Comment (thapav) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: Yes i crosschecked, i think we dont need it for memsistreams

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 18 above)
Comment (thapav) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: Yes i crosschecked, i think we dont need it for memsistream

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 22 above)
Comment (thapav) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: yes will change that

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 24 above)
Comment (thapav) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: yes , will check that

**Comment by thapav** [comment]
modified as per review comments, please check

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 26]
```diff
@@ -1961,6 +1965,368 @@ static void libc_stdio_setvbuf_tc(void)
...
+* @testcase         libc_stdio_nulloutstream_tc
+* @brief            Initializes a NULL stream. The initialized stream will write all data to the bit-bucket
+* @scenario         Initializes a NULL stream. The initialized stream will write all data to the bit-bucket
+* @apicovered       lib_nulloutstream
+* @precondition     NA
+* @postcondition    NA
+*/
+static void libc_stdio_nulloutstream_tc(void)
+{
+	FAR struct lib_outstream_s *stream;
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: Why does it need here?

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 27]
```diff
@@ -1961,6 +1965,368 @@ static void libc_stdio_setvbuf_tc(void)
...
+* @testcase         libc_stdio_nullinstream_tc
+* @brief            Initializes a NULL stream. The initialized stream will return only EOF
+* @scenario         Initializes a NULL stream. The initialized stream will return only EOF
+* @apicovered       lib_nullinstream
+* @precondition     NA
+* @postcondition    NA
+*/
+static void libc_stdio_nullinstream_tc(void)
+{
+	FAR struct lib_instream_s *stream;
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: Why does it need here? it is not used anywhere.

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 28]
```diff
@@ -1961,6 +1965,368 @@ static void libc_stdio_setvbuf_tc(void)
...
+*/
+static void libc_stdio_rawinstream_tc(void)
+{
+	int fd;
+	char *filename = VFS_FILE_PATH;
+
+	struct lib_rawinstream_s rawinstream;
+
+	fd = open(filename, O_RDONLY);
+	TC_ASSERT_NEQ("open", fd, OK);
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: This is wrong because you should check whether fd is not greater than equal to 0.
So It should be TC_ASSERT_GEQ("open", fd, 0);

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 29]
```diff
@@ -1961,6 +1965,368 @@ static void libc_stdio_setvbuf_tc(void)
...
+*/
+static void libc_stdio_rawoutstream_tc(void)
+{
+	int fd;
+	char *filename = VFS_FILE_PATH;
+
+	struct lib_rawoutstream_s rawoutstream;
+
+	fd = open(filename, O_RDONLY);
+	TC_ASSERT_NEQ("open", fd, OK);
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: This is wrong because you should check whether fd is not greater than equal to 0.
So It should be TC_ASSERT_GEQ("open", fd, 0);

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 30]
```diff
@@ -1961,6 +1965,368 @@ static void libc_stdio_setvbuf_tc(void)
...
+*/
+static void libc_stdio_rawsistream_tc(void)
+{
+	int fd;
+	char *filename = VFS_FILE_PATH;
+
+	struct lib_rawsistream_s rawsistream;
+
+	fd = open(filename, O_RDONLY);
+	TC_ASSERT_NEQ("open", fd, OK);
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: This is wrong because you should check whether fd is not greater than equal to 0.
So It should be TC_ASSERT_GEQ("open", fd, 0);

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 31]
```diff
@@ -1961,6 +1965,368 @@ static void libc_stdio_setvbuf_tc(void)
...
+
+	lib_rawsistream((FAR struct lib_rawsistream_s *)&rawsistream, fd);
+	TC_ASSERT_EQ("lib_rawsistream", rawsistream.fd, fd);
+	TC_ASSERT_EQ_CLEANUP("lib_rawsistream", rawsistream.public.nget, 0, close(fd));
+
+	lib_rawsistream((FAR struct lib_rawsistream_s *)&rawsistream, (fd + 2));
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: This is not meaningful unlike libc_stdio_memsistream_tc because fd is just integer value.
So these can be removed.

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 32]
```diff
@@ -1961,6 +1965,368 @@ static void libc_stdio_setvbuf_tc(void)
...
+	struct lib_rawsostream_s rawsostream;
+
+	fd = open(filename, O_RDONLY);
+	TC_ASSERT_NEQ("open", fd, OK);
+
+	lib_rawsostream((FAR struct lib_rawsostream_s *)&rawsostream, fd);
+	TC_ASSERT_EQ("lib_rawsostream", rawsostream.fd, fd);
+	TC_ASSERT_EQ_CLEANUP("lib_rawsostream", rawsostream.public.nput, 0, close(fd));
+
+	lib_rawsostream((FAR struct lib_rawsostream_s *)&rawsostream, (fd + 4));
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: Same above

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 33]
```diff
@@ -1961,6 +1965,368 @@ static void libc_stdio_setvbuf_tc(void)
...
+/**
+* @testcase         libc_stdio_rawsostream_tc
+* @brief            Initializes a stream for use with a file descriptor
+* @scenario         Initializes a stream for use with a file descriptor
+* @apicovered       lib_rawsostream
+* @precondition     NA
+* @postcondition    NA
+*/
+static void libc_stdio_rawsostream_tc(void)
+{
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: same above

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 34]
```diff
@@ -1961,6 +1965,368 @@ static void libc_stdio_setvbuf_tc(void)
...
+* @apicovered       lib_nullinstream
+* @precondition     NA
+* @postcondition    NA
+*/
+static void libc_stdio_nullinstream_tc(void)
+{
+	FAR struct lib_instream_s *stream;
+	struct lib_instream_s nullinstream;
+
+	lib_nullinstream(&nullinstream);
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: You have to add typecasting for lib_instream_s or put stream here

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 31 above)
Comment (thapav) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: done

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 30 above)
Comment (thapav) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: done

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 29 above)
Comment (thapav) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: done

**Comment by thapav** [comment]
modified as per review comments. please check

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 35]
```diff
@@ -1961,6 +1965,358 @@ static void libc_stdio_setvbuf_tc(void)
...
+*/
+static void libc_stdio_rawinstream_tc(void)
+{
+	int fd;
+	char *filename = VFS_FILE_PATH;
+
+	struct lib_rawinstream_s rawinstream;
+
+	fd = open(filename, O_RDONLY);
+	TC_ASSERT_GEQ("open", fd, OK);
```
Comment (jeongarmy) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: please use 0 instead of OK for checking fd
